scripts/utils/pre-release.mjs

```diff
@@ -0,0 +1,272 @@
+      choices: [
+        'Major',
+        'Minor',
+        'Patch',
+        'Custom',
+      ],
+      filter: value => value.toLowerCase(),
+    },
+    {
+      type: 'input',
+      name: 'customVersion',
+      message: 'Enter the custom version number.',
+      when: answers => answers.changeType === 'custom',
+      validate: (value) => {
+        if (!!semver.valid(value)) {
+          return true;
+        }
+
+        return 'The provided version is not a proper semver version number.';
+      },
+    },
+    {
+      type: 'input',
+      name: 'releaseDate',
+      message: 'Enter the release date in a form of DD/MM/YYYY.',
+      validate: (value) => {
+        const releaseDateValidity = validateReleaseDate(value);
+
+        if (releaseDateValidity.valid) {
+          return true;
+        }
+
+        return releaseDateValidity.error;
+      },
+    }
+  ];
+  const getConfirmationQuestion = (newVersion, formattedDate) => [
+    {
+      type: 'confirm',
+      name: 'isReleaseDateConfirmed',
+      message: `
+
+* New version: ${newVersion}
+* Release date: ${formattedDate}
+
+Are the version number and release date above correct?`,
+      default: true,
+    },
+  ];
+
+  // Version and release date were passed as arguments.
+  if (version && releaseDate) {
+    const semverBumpOptions = ['major', 'minor', 'patch'];
+    const releaseDateValidity = validateReleaseDate(releaseDate);
+    let newVersion = '';
+
+    if (!releaseDateValidity.valid) {
+      displayErrorMessage(releaseDateValidity.error);
+      process.exit(1);
+    }
+
+    if (isVersionValid(version)) {
+      newVersion = version;
+
+    } else if (semverBumpOptions.includes(version)) {
+      newVersion = semver.inc(currentVersion, version);
+
+    } else {
+      displayErrorMessage(
+        `${version} is not a valid version number, nor a semver change type (major/minor/patch).`);
+      process.exit(1);
+    }
+
+    if (!newVersion) {
+      displayErrorMessage(`Something went wrong while updating the version number with semver.`);
+      process.exit(1);
+    }
+
+    displayConfirmationMessage(
+      `\nChanging the version number to ${newVersion}, to be released on ${releaseDate}. \n`
+    );
+
+    setVersion(newVersion, workspacePackages);
+    setReleaseDate(releaseDate);
+
+  } else {
+    await inquirer.prompt(questions).then(async (answers) => {
+      const releaseDateObj = moment(answers.releaseDate, 'DD/MM/YYYY', true);
+
+      const newVersion =
+        answers.changeType !== 'custom' ?
+          getVersionFromReleaseType(answers.changeType, currentVersion) :
+          answers.customVersion;
+
+      await inquirer.prompt(
+        getConfirmationQuestion(newVersion, releaseDateObj.format('DD MMMM YYYY'))
+      ).then((confirmationAnswers) => {
+        if (confirmationAnswers.isReleaseDateConfirmed) {
+          setVersion(newVersion);
+          setReleaseDate(answers.releaseDate);
+        }
+      });
+    });
```

Comment:
I think that we can take advantage of async/await and transform the code to make appear more synchronously. Thanks to that we get rid of `then` calls.  
```suggestion
    const answers = await inquirer.prompt(questions);
    const releaseDateObj = moment(answers.releaseDate, 'DD/MM/YYYY', true);

    const newVersion =
      answers.changeType !== 'custom' ?
        getVersionFromReleaseType(answers.changeType, currentVersion) :
        answers.customVersion;

    const confirmationAnswers = await inquirer.prompt(
      getConfirmationQuestion(newVersion, releaseDateObj.format('DD MMMM YYYY')));

    if (confirmationAnswers.isReleaseDateConfirmed) {
      setVersion(newVersion);
      setReleaseDate(answers.releaseDate);
    }
```